c7n/schema.py

```diff
@@ -578,6 +579,83 @@ def resource_vocabulary(cloud_name=None, qualify_name=True, aliases=True):
     return vocabulary
 
 
+class ExpandedSchemaMeta(type):
+    """Expand an element's schema using service model shape data
+
+        Repurpose some of the shape discovery/validation logic in
+        c7n.resources.aws.shape_validate() to dynamically expand
+        element schema using the latest service model shape information.
+
+        Include available properties, their types, and enumerations of
+        possible values where available.  Assumes the use of type_schema().
+
+        Attributes:
+            service (str): The AWS service for the element. (required)
+            shape_name (str): The service model request shape name. (required)
+            attributes_key (str): Some Custodian actions use a top-level
```

Comment:
I'm not seeing attributes_key usage in the way described, extant usage seems to be a way to spell an annotation key.